gallery-so/gallery
# Gallery Editor v0

### Summary of Changes

- Added gallery editor feature 
- Introduce new component `<SortableRowList/>` to handle the dnd for our row! This is where all the logic for dnd happens
- The editor was virtualized
- User able to move section up and down
- User able to move row in the **same section**
- User able to add new token into **selected row**
- All the changes can be saved and reflected on the web
- Moved most of **web** logic to the `/shared`

### Demo or Before/After Pics

https://github.com/gallery-so/gallery/assets/4480258/9b369085-01de-4fd7-880d-62d3e027e3e0




### Edge Cases

1. Add new token into the row
2. Move section untill the top or bottom of the screen
3. Render a row although the column not fulfilled. E.g: Column 3 but only have 1 token

### Testing Steps

1. Go to **galleries** tab in the **profile** screen
2. You should see 3 dots in each gallery to edit gallery (hide behind the feature flag)

### Checklist

Please make sure to review and check all of the following:
